mozetl/hardware_report/check_output.py

```diff
@@ -0,0 +1,61 @@
+import ujson
+import re
+
+from mozetl.utils import send_ses
+
+
+def check_output():
+    data = _get_data()
+    changes = _check_most_recent_change(data, min_change=3)
+
+    if len(changes > 0):
+        _report_changes(changes)
+
+
+def _get_data():
+    """
+    Make the date values the keys for each json blob.
+
+    For example:
+    Transform [ {date: 20170101, data1: some1, ...}, {date:20170702, data1:some2}, ... ]
+    into { 20170101: {data1: some1, ...}, 20170102: {data1:some2, ...} }
+    """
+    with open("hwsurvey-weekly.json", "r") as hwsurvey:
+        data = ujson.loads(''.join(hwsurvey.readlines()))
+
+    return {int(re.sub('-', '', e['date'])): {k: e[k] for k in e if k != 'date'} for e in data}
+
+
+def _check_most_recent_change(values, min_change=.05, min_value=0.01, missing_val=.01):
+    recent_week = max(values.keys())
+    second_recent_week = max(values.viewkeys() - {recent_week})
+
+    base, compare = values[second_recent_week], values[recent_week]
+    changes = [
+        (k, (compare.get(k, missing_val) - base.get(k, missing_val)) / base.get(k, missing_val))
```

Comment:
Is zero division possible here (in practice)? Maybe we should break this comparison out into a separate function or filter to minimum base size before calculating the change.

Reply:
Totally optional: using `(compare/base) - 1` may be more succinct.